lib/pages/timetable/service/uni_event_provider.dart

```diff
@@ -272,6 +272,19 @@ class UniEventProvider extends EventProvider<UniEventInstance>
               }
               if (data['teacher'] != null) {
                 teacher = await _personProvider.fetchPerson(data['teacher']);
+                final String teacherId = await getTeacherDocumentId(teacher);
```

Comment:
See above, we shouldn't save this field in the class document.
Plus, this approach is woefully inefficient - every time we fetch events, we also make multiple write operations on the classes document? Not to mention saving both the ID and the name doesn't make sense either, since you can find the ID based on the name anyway.